ui/pages/login/redirect.js

```diff
@@ -0,0 +1,71 @@
+import { useEffect } from 'react'
+import { useRouter } from 'next/router'
+import Cookies from 'universal-cookie'
+
+import { currentBaseDomain } from '.'
+import LoginLayout from '../../components/layouts/login'
+
+export default function Redirect() {
+  const router = useRouter()
+  const { isReady } = router
+  const { code, state } = router.query
+
+  useEffect(() => {
+    async function finish({ code, state }) {
+      const cookies = new Cookies()
+      // start with the base of the login callback URL we will redirect to
+      let callbackURL =
+        currentBaseDomain() +
+        '/login/callback' +
+        '?code=' +
+        code +
+        '&state=' +
+        state
+      let org = cookies.get('finishLogin') // the org to redirect to is stored in this cookie
+      if (org !== undefined && org !== '') {
+        // build the callback URL to finish the login at the org
+        callbackURL = org + '.' + callbackURL
+        // login redirect is complete so we no longer need this cookie
+        cookies.remove('finishLogin', {
+          path: '/',
+          domain: `.${currentBaseDomain()}`,
+        })
+      }
+      callbackURL = window.location.protocol + '//' + callbackURL
+      // send the browser to the org specific callback URL to finish login
+      router.replace(callbackURL)
+    }
+
+    if (code && state) {
+      finish({ code, state })
+    }
+  }, [code, state, router])
+
+  if (!isReady) {
+    return null
+  }
+
+  return (
+    <div className='my-32 flex h-full w-full items-center justify-center'>
+      <svg
```

Comment:
is this a loader image? We have a loader component call `<Loader />` can we use that?

Reply:
Nice, using this component now